test/unittests/parser_test.cpp

```diff
@@ -1241,10 +1243,23 @@ TEST(parser, data_section_memidx_nonzero)
 {
     const auto section_contents = make_vec({"0141010b0100"_bytes});
     const auto bin = bytes{wasm_prefix} + make_section(11, section_contents);
-
     EXPECT_THROW_MESSAGE(parse(bin), parser_error, "unexpected memidx value 1");
 }
 
+TEST(parser, DISABLED_empty_data_section_without_memory)
+{
+    const auto bin = bytes{wasm_prefix} + make_section(11, make_vec({}));
```

Comment:
So I think this test could be changed to having empty data vector
```suggestion
    const auto bin = bytes{wasm_prefix} + make_section(11, make_vec({0041010b00}));
```